ReactiveCocoa/Swift/Property.swift

```diff
@@ -15,6 +15,37 @@ public protocol PropertyType {
 	var signal: Signal<Value, NoError> { get }
 }
 
+extension PropertyType {
+	/// Maps each value in the property to a new value.
+	@warn_unused_result
+	public func map<U>(transform: Value -> U) -> AnyProperty<U> {
+		return AnyProperty(
+			initialValue: transform(value),
+			signal: signal.map(transform)
```

Comment:
`MutableProperty.producer` should be more appropriate here, shouldn't it? Since there is no umbrella lock here (and you can't access `withValue(action:)` here either), the signal could have pushed a new value before the `AnyProperty` eventually subscribed to the signal.